### ISSUE TITLE
Brick port vs gluster client port

### ISSUE THREAD

**Issue by bronislav** [gluster/glusterd2#1288]
What ports should be exposed to the mount clients?

From the network documentation, it says that only sunRPC port should be exposed to the clients. Making me think that this is the only port required to be able to mount gluster volume. But I'm unable to do so.

It seems to me from logs that gluster client trying to connect to the brick port after receiving volume file. Is it correct behavior?

Do I need to open brick port as well?

**Comment by amarts** [comment]
@bronislav yes, clients always get the correct brick port from glusterd (or glusterd2) process which runs in port 24007, and then reconnects to the correct port.

Clients can use any port range, and generally uses one of the free ports from `bind(3)` systemcall.

**Comment by bronislav** [comment]
@amarts One more question.

I've set port range with sysctl: 
```
sh-4.2# sysctl net.ipv4.ip_local_port_range
net.ipv4.ip_local_port_range = 29003	29004
```

Does it mean that gluster brick should pick 29003 port? It still chooses 49152 port instead of above.

**Comment by prashanthpai** [comment]
@bronislav Are you using glusterd2 at all ? If yes, what version of glusterd2 are you using ?

`glusterd2 --version` will give you the version information.

**Comment by bronislav** [comment]
@prashanthpai 
```
sh-4.2# glusterd2 --version
glusterd version: v4.1.0-202.gitea0c111
git SHA: ea0c111
go version: go1.9.4
go OS/arch: linux/amd64
sh-4.2#
```

**Comment by bronislav** [comment]
It is the latest available build, but it is quite old:
http://artifacts.ci.centos.org/gluster/gd2-nightly/master/7/x86_64/

**Comment by prashanthpai** [comment]
> @prashanthpai
> 
> ```
> sh-4.2# glusterd2 --version
> glusterd version: v4.1.0-202.gitea0c111
> git SHA: ea0c111
> go version: go1.9.4
> go OS/arch: linux/amd64
> sh-4.2#
> ```
> It looks like it is very old version. Going to dig why it is not recent. I'm installing from nightly repos.

From the git SHA version, I can see that the enhancement (commit a00ca91d) you are looking for isn't present in the version you're using. Please try with the latest glusterd2 build.

**Comment by bronislav** [comment]
@prashanthpai I figured this out already. But the version that I am using is the latest available in the nightly repo http://artifacts.ci.centos.org/gluster/gd2-nightly/master/7/x86_64/

Where I can find latest RPMs?

**Comment by bronislav** [comment]
@phlogistonjohn RMP build job is failing since Sep 18 https://ci.centos.org/job/gluster_gd2-nightly-rpms/

May I help with anything to get it fixed?

**Comment by prashanthpai** [comment]
Indeed, just found out the same.
@kshlm Can you please take a look ?

**Comment by bronislav** [comment]
@prashanthpai @kshlm Any chance to get it fixed soon?

**Comment by prashanthpai** [comment]
@kshlm is on it.

**Comment by atinmu** [comment]
If @kshlm is yet to look into it and considering his plate is full, I'd request if @vpandey-RH or @rishubhjain one of you can take a look at this and address it asap. Having nightly build failures for days isn't a good sign.

**Comment by prashanthpai** [comment]
@bronislav While it's getting fixed, feel free to download the [latest binaries](https://github.com/gluster/glusterd2/releases) (just 3 days old) that contain the enhancement.

**Comment by kshlm** [comment]
#1298 should fix the nightly builds.

**Comment by atinmu** [comment]
Closed through #1298
